src/DynamoPackages/PackageLoader.cs

```diff
@@ -418,16 +422,50 @@ public Package ScanPackageDirectory(string directory, bool checkCertificates)
                 // prevent loading unsigned packages if the certificates are required on package dlls
                 if (checkCertificates)
                 {
-                    CheckPackageNodeLibraryCertificates(directory, discoveredPkg);
+                    CheckPackageNodeLibraryCertificates(directory, discoveredPackage);
                 }
 
-                // prevent duplicates
-                if (LocalPackages.All(pkg => pkg.Name != discoveredPkg.Name))
+                var discoveredVersion = CheckAndGetPackageVersion(discoveredPackage.VersionName, discoveredPackage.Name, discoveredPackage.RootDirectory);
+
+                var existingPackage = LocalPackages.FirstOrDefault(package => package.Name == discoveredPackage.Name);
+
+                // Is this a new package?
+                if (existingPackage == null)
+                {
+                    // Yes
+                    Add(discoveredPackage);
+                    return discoveredPackage; // success
+                }
+
+                var existingVersion = CheckAndGetPackageVersion(existingPackage.VersionName, existingPackage.Name, existingPackage.RootDirectory);
+
+                // Is this a duplicated package?
+                if (existingVersion == discoveredVersion)
+                {
+                    // Duplicated with the same version
+                    throw new LibraryLoadFailedException(directory,
+                        String.Format(Properties.Resources.DulicatedPackage,
+                            discoveredPackage.Name,
+                            discoveredPackage.RootDirectory));
+                } 
+                
+                // Is the existing version newer?
+                if (existingVersion > discoveredVersion)
                 {
-                    Add(discoveredPkg);
-                    return discoveredPkg; // success
+                    // Older version found, show notification
+                    throw new LibraryLoadFailedException(directory, String.Format(Properties.Resources.DuplicatedOlderPackage,
+                            existingPackage.Name,
+                            discoveredPackage.RootDirectory,
+                            existingVersion.ToString(),
+                            discoveredVersion.ToString()));
                 }
-                throw new LibraryLoadFailedException(directory, String.Format(Properties.Resources.DulicatedPackage, discoveredPkg.Name, discoveredPkg.RootDirectory));
+
+                // Newer version found, show notification.
+                throw new LibraryLoadFailedException(directory, String.Format(Properties.Resources.DuplicatedNewerPackage,
```

Comment:
I still don't understand this case. Are we rejecting the new version over here and keeping the old one? If so, why?

Reply:
Yes, this is the where we keep the old one and reject the new one. This was the major change I made after the feedback yesterday. Instead of using the new version we just inform the user that we have found a newer version that we will ignore because we have already found another version of the package. That should give them enough information so they can decide  what directory order they want.

Reply:
How about raising the existing `mark for uninstall` dialog here asking the user if they'd like to mark the existing package for uninstall and install the newer version upon restart or to simply ignore the new version?

Reply:
Or does that not make sense at this stage?

Reply:
I think we should wait with that for now and add that as a separate task if we need it.

Reply:
I agree...save it for later. Adding msg boxes when starting up dynamo could be very impactful to UX.

Reply:
That's okay, but just to clarify, I don't think this case is at startup, it's at the time you add a package path if I'm not wrong.

Reply:
It can happen at startup. Package X version 1.0.0 is installed in standard library. User download version 2.0.0. The user will get the message above every time they start after that as long as the standard library is first in the directory list. That is one of those cases when the user can't uninstall the old package because most of them will not have write access to the standard library. So lets see how it plays out and go from there.

Reply:
I doubt there's gonna be a case where we deliver a package in the std lib with a version that's older than that available on the PM, since that's the decision that we're making as a team, but I agree, it's not impossible. It could happen with host integrator packages, etc.

Reply:
I can imagine we might create a new version of `meshtoolkit` say for release to PM to get early feedback before incorporation into std.lib.

Reply:
🙏